socorro/collector/submitter_app.py

```diff
@@ -313,40 +263,52 @@ def source_iterator(self):
 
     #--------------------------------------------------------------------------
     def _infinite_iterator(self):
+        """the default iterators will yield None objects if there is nothing
+        in the queue.  The submitter doesn't want those, so it declines to
+        to yield them"""
         while True:
-            for crash_id in self.new_crash_source.new_crashes():
-                yield crash_id
-
-    #--------------------------------------------------------------------------
-    def _all_iterator(self):
-        for crash_id in self.new_crash_source.new_crashes():
-            yield crash_id
+            for crash_id in self._basic_iterator():
+                if crash_id is not None:
+                    yield crash_id
 
     #--------------------------------------------------------------------------
     def _limited_iterator(self):
```

Comment:
How is this different from it's parent class? If it is, it should be explained in a doc string or a comment like you did above in `SubmitterApp._infinite_iterator`.